multimethod/__init__.py

```diff
@@ -92,11 +92,18 @@ def subcheck(cls, tp):
         )
 
 
+def _is_metaclass(cls):
+    try:
+        return issubclass(cls, type)
+    except TypeError:
+        return False
+
+
 def distance(cls, subclass):
     """Return estimated distance between classes for tie-breaking."""
     if getattr(cls, '__origin__', None) is typing.Union:
         return min(distance(arg, subclass) for arg in cls.__args__)
-    mro = subclass.mro()
+    mro = subclass.mro(subclass) if _is_metaclass(subclass) else subclass.mro()
```

Comment:
It can be called as an unbound method. So I think I would prefer `mro = type.mro(subclass)`.
```python
In []: type.mro(enum.EnumMeta)
Out[]: [enum.EnumMeta, type, object]

In []: type.mro(bool)
Out[]: [bool, int, object]
```

Reply:
Initially I used `type.mro(subclass)`, but then I thought it's possible that some `type` subclasses might override this method in some meaningful way?

E.g.

```python
 class MetaMock(type):
    def mro(mcs):
        # lie about this metaclass; say it's just the plain `type`
        return type, object
```
then `type.mro(MetaMock)` would yield `(MetaMock, type, object)` and `MetaMock.mro(MetaMock)` would yield `(type, object)`.

Not sure if this is "allowed" or if `multimethod` should believe such a lie or use the true MRO. Maybe a better example would be a special implementation of `mro` that is faster or more appropriate than the default `type.mro` implementation, but I don't have any such example in mind.

Reply:
I don't think there's a use case here, because it underscores how rare it would be to dispatch on types themselves. A custom `mro` affects the `issubclass` relation of _its_ types, and multimethod further uses `mro` explicitly to break ties.

Whereas a custom `mro` of a _metaclass_ - essentially a `staticmethod` on it - just defies description. This example isn't usable:
```python
In []: class Mock(metaclass=MetaMock):
    ...:     pass

TypeError: mro() returned base with unsuitable layout ('type')
```

Reply:
Right; I've reverted to `type.mro`.